lang/cs/Org.Apache.REEF.Tests/Functional/Failure/User/TaskConstructorExceptionTest.cs

```diff
@@ -0,0 +1,175 @@
+                var contextConf = ContextConfiguration.ConfigurationModule
+                    .Set(ContextConfiguration.Identifier, ContextId)
+                    .Build();
+
+                var taskConf = TaskConfiguration.ConfigurationModule
+                    .Set(TaskConfiguration.Identifier, TaskId)
+                    .Set(TaskConfiguration.Task, GenericType<TestConstructorExceptionTask>.Class)
+                    .Build();
+
+                value.SubmitContextAndTask(contextConf, taskConf);
+            }
+
+            public void OnNext(IFailedTask value)
+            {
+                var taskClientCodeEx = value.AsError() as TaskClientCodeException;
+                if (taskClientCodeEx == null)
+                {
+                    throw new Exception("Expected Exception to be a TaskClientCodeException.");
+                }
+
+                if (taskClientCodeEx.ContextId != ContextId)
+                {
+                    throw new Exception("Expected Context ID to be " + ContextId + ", but instead got " + taskClientCodeEx.ContextId);
+                }
+
+                if (taskClientCodeEx.TaskId != TaskId)
+                {
+                    throw new Exception("Expected Task ID to be " + TaskId + ", but instead got " + taskClientCodeEx.TaskId);
+                }
+
+                Exception error = taskClientCodeEx;
+
+                var foundErrorMessage = false;
+                while (error != null)
+                {
+                    // Using Contains because the Exception may not be serializable
+                    // and the message of the wrapping Exception may be expanded to include more details.
+                    if (error.Message.Contains(TestConstructorExceptionTask.ConstructorErrorMessage))
+                    {
+                        foundErrorMessage = true;
+                        break;
+                    }
+
+                    error = error.InnerException;
+                }
+
+                if (!foundErrorMessage)
+                {
+                    throw new Exception("Expected to find error message " + 
+                        TestConstructorExceptionTask.ConstructorErrorMessage + " in the layer of Exceptions.");
+                }
+
+                Logger.Log(Level.Info, ReceivedFailedTaskEvent);
+                value.GetActiveContext().Value.Dispose();
+            }
+
+            public void OnError(Exception error)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void OnCompleted()
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private sealed class TestConstructorExceptionTask : ITask
+        {
+            public const string ConstructorErrorMessage = "ConstructorErrorMessage";
+
+            [Inject]
+            private TestConstructorExceptionTask()
+            {
+                throw new Exception(ConstructorErrorMessage);
```

Comment:
Yes, this test is specifically to catch the Exception thrown from the constructor. The change in this PR only covers the cases where the Task Exception is thrown upon Task Start. As explained in my other comment, `TaskClientCodeException` can only be thrown on control flow errors.